### ISSUE TITLE
DatePicker before 1970 returning month offset by 1

### ISSUE THREAD

**Issue by davideme** [marmelab/admin-on-rest#1189]
**What you were expecting:**
Be able to select a date on march on any year before 1970

**What happened instead:**
When displaying any month(except January and February) with a DatePicker, and selecting a date it selecting the next one.

**Steps to reproduce:**

- Go to the demo (https://marmelab.com/admin-on-rest-demo/#/customers/121)
- Edit a customer
- Select a date before 1970
- The same date with an offset by one month is selected

<img width="1527" alt="screen shot 2017-10-30 at 19 29 15" src="https://user-images.githubusercontent.com/858090/32188697-d4b9ff22-bda8-11e7-833d-193a304753a3.png">

**Other information:**
Reproducible in the Demo and our local admin on rest env.

**Environment**

* Admin-on-rest version:1.3.2
* React version: 15.5.4
* Browser: Chrome

**Comment by fzaninotto** [comment]
This is likely a material-ui bug, please open an issue on their bug tracker: https://github.com/callemall/material-ui/issues

**Comment by pedrovgs** [comment]
@fzaninotto the Material UI date picker is working properly, it seems to be related with the configuration in admin-on-rest for the component ``DateInput``.

**Comment by fzaninotto** [comment]
OK then, reopening.

**Comment by fzaninotto** [comment]
This bug is 1.x only, the `next` branch simply doesn't have a datepicker.

**Comment by afilp** [comment]
@fzaninotto How can we pick dates in 'next' branch? Thanks!

**Comment by fzaninotto** [comment]
Using the native browser datepicker:

![image](https://user-images.githubusercontent.com/99944/34573706-52fb7aaa-f175-11e7-96e3-9eb9d7256077.png)

**Comment by djhi** [comment]
The MUI DatePicker does have the same issue. It's related to the `DateTimeFormat` prop.

Checkout the MUI the *Localized example* in the [documentation](http://www.material-ui.com/#/components/date-picker).

It was fixed by https://github.com/mui-org/material-ui/pull/7056 which has been reverted.

The only option would be to not use the `DateTimeFormat` prop but I'm not sure about the impact.

Ping @fzaninotto

**Comment by fzaninotto** [comment]
You'll have to live with this bug until MUI fixes it - or upgrade to admin-on-rest 2.0, when it's released.

**Comment by djhi** [comment]
It's not a bug in MUI either. It's related to the way `DateTimeFormat` works.

Workaround: pass your own `DateTimeFormat` in `options`:

```jsx
<DateInput
    source="created_at"
    options={{ DateTimeFormat: undefined }}
/>
```
